### PR TITLE
refactor: upgrade to rxjs 6 and ts 2.7

### LINKED ISSUE #926: ngrx V6: upgrade to rxjs 6 and ts2.7
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[x] Feature request
[ ] Documentation issue or request
</code></pre>

## What is the current behavior?

Could not compile app with ngrx, Angular v6.0.0-rc.0, rxjs and rxjs-compat v6.0.0-beta.1

```bash
ERROR in ../node_modules/@ngrx/effects/@ngrx/effects.es5.js
Module not found: Error: Can't resolve 'rxjs/operator/exhaustMap' in ...
```

## Expected behavior:

I hope I could upgrade my app to Angular v6 by the time it releases. But I'm using a lot of ngrx in my code and it seems not compatible with Angular v6 and ts 2.7 yet.

## Minimal reproduction of the problem with instructions:

Create an new cli project and run `ng update --next`
then run `ng build`

## Version of affected browser(s),operating system(s), npm, node and ngrx:
ngrx v5.2.0, node 8.10.0, yarn 1.5.1
## Other information:
Please tell me if I could help. I'm willing to do a PR.

### PR DESCRIPTION
**Pull request by sandangel**
Closes #926 , #940 , #941

### CODE REVIEW

File: modules/router-store/src/router_store_module.ts [hunk 1]
```diff
@@ -283,7 +284,7 @@ export class StoreRouterConnectingModule {
   }
 
   private setUpStateRollbackEvents(): void {
-    this.router.events.subscribe(e => {
+    this.router.events.subscribe((e: RouterEvent) => {
```
Comment (MattiJarvinen-BA) [inline modules/router-store/src/router_store_module.ts]: VS Code shows correct typing completions for:

``` typescript
this.router.events.subscribe( e => { } );
```

I don't think adding typing here has any real value then again there's no real harm either.

Comment (sandangel) [reply modules/router-store/src/router_store_module.ts]: yes, but CI compile was throw err. My VS Code complain about this too.

File: docs/effects/api.md [hunk 2]
```diff
@@ -5,35 +5,29 @@
...
+
 ```ts
 @NgModule({
-  imports: [
-    EffectsModule.forFeature([
-      SomeEffectsClass,
-      AnotherEffectsClass,
-    ])
-  ]
+  imports: [EffectsModule.forFeature([SomeEffectsClass, AnotherEffectsClass])],
```
Comment (MattiJarvinen-BA) [inline docs/effects/api.md]: Previous formatting was way more readable. This is read me after all.

Comment (sandangel) [reply docs/effects/api.md]: This was because of husky and prettier auto run after I commit

Comment (tdeschryver) [reply docs/effects/api.md]: If needed, this you can use `prettier-ignore` to ignore a line/file/...

Comment (sandangel) [reply docs/effects/api.md]: I think changing format setup is not the main purpose of this PR, I just used the config that has been set. If brandon or mike suggest it I will change it accordingly.

File: modules/router-store/src/router_store_module.ts (hunk 1 above)
Comment (sandangel) [reply modules/router-store/src/router_store_module.ts]: I think this was because of using angular v5. The error has gone when I upgrade to ng v6

**Comment by brandonroberts** [comment]
@sandangel jasmine-marbles `0.3.0` has been released with support for RxJS 6

**Comment by sandangel** [comment]
cool, I will upgrade it now. Currently this project can not be compile with angular v6 rc0. I hope the angular team will release rc1 today.

**Comment by sandangel** [comment]
@brandonroberts Hi, look like there is a bug with jasmine-marbles, could you help me take a look?

**Comment by brandonroberts** [comment]
@sandangel sure. I'll pull down your changes. If you want to discuss we can meet in the [gitter channel](https://gitter.im/ngrx/platform)

**Comment by brandonroberts** [comment]
Yep. Looking now

**Comment by brandonroberts** [comment]
@sandangel the schematics package has been updated to support workspaces, and our repo isn't setup to match that structure yet. I generated a new project from the latest CLI beta here for reference https://github.com/brandonroberts/ngrx-scaling-train

Since those changes are outside the scope of these updates, I'd say disable the component schematics tests for now.

The router test is a little more difficult since it seems to be a change in behavior with the latest version of RxJS that I'm still digging into. I think that one should be disabled for now also because under normal circumstances you wouldn't throw an error from a reducer.

**Comment by sandangel** [comment]
Hi @brandonroberts , thank you for your help.
With these changes I think this PR is ready for review.

**Comment by brandonroberts** [comment]
@sandangel cool! Will you rebase this branch on top of the `next` branch and fix the merge conflicts so we only see the latest changes?

**Comment by sandangel** [comment]
@brandonroberts sure I will

**Comment by MikeRyanDev** [comment]
:clap: :clap: :clap:

**Pull request merged**